### PR TITLE
Allows cyborgs to quicken their self untip time with roleplay

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request
image
(halves their untip timer when they emote, only works once)

## Why It's Good For The Game
borgs should be encouraged for roleplaying their anger at being tipped

## Changelog

:cl:
balance: Borgs can halve their selfuntip time with the power of roleplay.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
"emote" is a bit to meta

how about "frustration" has empowered you

**Comment by SuperNovaa41** [comment]
> "emote" is a bit to meta
> 
> how about "frustration" has empowered you

good idea

File: code/datums/components/tippable.dm [hunk 1]
```diff
@@ -209,3 +223,19 @@
...
+	if (roleplayed)
+		return
+	if (!is_type_in_list(emote, roleplay_emotes))
+		return
+	var/time_left = timeleft(self_untip_timer)
+	deltimer(self_untip_timer)
+	self_untip_timer = addtimer(CALLBACK(src, .proc/right_self, user), time_left / 2, TIMER_UNIQUE | TIMER_STOPPABLE)
+	roleplayed = TRUE
+
+	to_chat(user, span_notice("Your frustration has empowered you! You can now right yourself faster!"))
```
Comment (tralezab) [inline code/datums/components/tippable.dm:241]: if we're doing this, this needs to be a callback that is invoked from this component and handled on the borg itself. this is too literal

Comment (SuperNovaa41) [reply code/datums/components/tippable.dm:241]: the entire proc does?

**Comment by Time-Green** [comment]
Should probably make it obvious this is a thing, it seems really obscure but gives a huge advantage

File: code/datums/components/tippable.dm [hunk 2]
```diff
@@ -209,3 +223,19 @@
...
+
+	if (!is_tipped)
+		return
+	if (roleplayed)
+		return
+	if (!is_type_in_list(emote, roleplay_emotes))
+		return
+	var/time_left = timeleft(self_untip_timer)
+	deltimer(self_untip_timer)
+	self_untip_timer = addtimer(CALLBACK(src, .proc/right_self, user), time_left / 2, TIMER_UNIQUE | TIMER_STOPPABLE)
```
Comment (MrDoomBringer) [inline code/datums/components/tippable.dm:238]:
```suggestion
	self_untip_timer = addtimer(CALLBACK(src, .proc/right_self, user), time_left * 0.8, TIMER_UNIQUE | TIMER_STOPPABLE)
```
not sure about this change, if we have it at all it should be less drastic an advantage. Either way we prefer multiplying by a decimal

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by SomeDongus** [comment]
> Should probably make it obvious this is a thing, it seems really obscure but gives a huge advantage

That's what they get for not checking the changelog

**Comment by BadIsMyUsername** [comment]
How about a tip assist upgrade/module for borgs?
https://www.youtube.com/watch?v=v5VLUbHYQes
Tip assist

File: code/datums/components/tippable.dm (hunk 2 above)
Comment (SuperNovaa41) [reply code/datums/components/tippable.dm]: sure, i changed it to .75

**Comment by tralezab** [comment]
you know what

**Comment by tralezab** [comment]
we may as well have both tipping and this, or neither

**Comment by Shadowflame909** [comment]
this is the start of the mime borg saga

**Pull request merged**